pkg/exasol-rest-api/integration_test.go

```diff
@@ -0,0 +1,207 @@
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - insufficient privileges: CREATE SESSION.\"}")
+}
+
+func (suite *IntegrationTestSuite) TestExasolUserWithWrongCredentials() {
+	router := suite.startServer(suite.createApplication(&exasol_rest_api.ApplicationProperties{
+		ExasolUser:                "not_existing_user",
+		ExasolPassword:            "wrong_password",
+		ExasolPort:                suite.exasolPort,
+		ExasolWebsocketApiVersion: 2,
+	}))
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - authentication failed.\"}")
+}
+
+func (suite *IntegrationTestSuite) TestWrongExasolPort() {
+	router := suite.startServer(suite.createApplication(&exasol_rest_api.ApplicationProperties{
+		ExasolUser:                suite.defaultExasolUsername,
+		ExasolPassword:            suite.defaultExasolPassword,
+		ExasolPort:                4321,
+		ExasolWebsocketApiVersion: 2,
+	}))
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(), "{\"ErrorCode\":\"EXA-REST-API-1\"")
+	suite.Contains(responseRecorder.Body.String(), "connect: connection refused")
+}
+
+func (suite *IntegrationTestSuite) TestWrongWebsocketApiVersion() {
+	router := suite.startServer(suite.createApplication(&exasol_rest_api.ApplicationProperties{
+		ExasolUser:                suite.defaultExasolUsername,
+		ExasolPassword:            suite.defaultExasolPassword,
+		ExasolPort:                suite.exasolPort,
+		ExasolWebsocketApiVersion: 0,
+	}))
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[00000] Could not create WebSocket protocol version 0\"}")
+}
+
+func runExasolContainer(ctx context.Context) testcontainers.Container {
+	request := testcontainers.ContainerRequest{
+		Image:        "exasol/docker-db:7.0.10",
```

Comment:
I have some problems with 7.1.1, the test is broken with it and I can't detect the problem right away. Maybe it's WebSockets version or something else. It will take time to investigate it, I'll do separatly

Reply:
https://github.com/exasol/exasol-rest-api/issues/14